### PR TITLE
17914 added popup for socieites

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity#17914

Added a popup for SO and XSO for name change and restoration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:106]: Please move this to its own component, same as the other dialogs in this project.

Comment (severinbeauvais) [inline src/components/new-request/search.vue:106]: I think you don't need the v-if here, because you are already controlling the visibility of this dialog with the v-model property.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:143]: Would it make more sense to move this check to the entire template block here (line 140)?

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:509]: You don't need this getter because your dialog is a fixed width.

In pick-entity-or-conversion.vue, the dialog could be really wide, ie

image

or narrower:

image

Comment (severinbeauvais) [reply src/components/new-request/search.vue:509]: Looks like a static value of 550px is correct for your dialog.

image

**Comment by severinbeauvais** [comment]
I have tested this with the "enable-society" FF off and the dialogs display correctly if I select a society for name change or restoration. Alteration is already protected by the mappings lists but I think it should be handled the same way as name change and restoration.

I tested this with the "enable-society" FF on and the name change flow works fine; the alteration flow is blocked by the mapping and should be fixed; the restoration flow works for extra societies (XS) but not BC societies (S) and should be fixed.

**Comment by severinbeauvais** [comment]
Do not merge this until after the release on Tuesday, October 31.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:245]: Please check with UI/UX team (eg, Janis or Yui) on whether they want to see "To request a restoration for a Society".

File: src/interfaces/new-request-interface.ts
Comment (severinbeauvais) [inline src/interfaces/new-request-interface.ts:70]: I had suggested just passing a prop to the dialog to show/hide it as that would be cleanest.

It's OK that you figured out how the other modals in Namerequest UI are show/hidden.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:106]: Remove this debugging code as some point in this PR.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Thanks.

Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Fixed, thanks.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue:452]: Fixed, thanks.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

**Comment by severinbeauvais** [comment]
I restarted all checks that failed due to environmental issue. They have now passed.

This flow still doesn't work. I should see more controls after this:

image

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue:452]: I have unresolved this conversation because you left in the `madWidth` getter that you had initially added and is no longer used. Please clean this up.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:514]: I'm quite confused by this getter (computed). It should not be doing multiple things (and esp not setting state when you "get" something).

I think you go back to my earlier suggestion of passing a prop to show/hide the dialog, and using an event handler when it's closed to clean up the variables.

After that, if you still need to call something when a variable changes then use a watcher method.

**Review by severinbeauvais — CHANGES_REQUESTED** [review]
Not OK as is. Please rework. Ask for more help or examples if needed.

PS You're close, and Namerequest UI is _frustratingly_ complicated, I know, so I harbour no ill feelings :)

**Comment by eve-git** [comment]
Could you please update the version number to 5.3.5, as https://github.com/bcgov/namerequest/pull/742 is currently using 5.3.4 and merged already.

File: src/components/dialogs/societies-info-dialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/societies-info-dialog.vue:47]: It's good practise to include a default value, or to make this prop required.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:105]: Looks good!

Please put each attribute on a different line. I don't know if that lint rule is used in this repo but we try to do that in the other UIs.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:445]: Looks good!

You can leave out `oldVal` if you don't need it.

I'd rename `newVal` to something more meaningful (like "disabled"?).

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:105]: You were asking about best practises for modals...

In this repo, all dialogs (except this one) are declared in App.vue and each one uses its own store getter + action to show/hide. Personally I don't like this approach because a lot of code files are involved; it's not "contained" or localized.

In the other repos, each dialog generally looks like this in the parent component:
```
<MyDialog
  :dialog="whetherToShowThisDialog"
  @close="closeThisDialog()"
/>

<script>
  ...
  whetherToShowThisDialog = false // initial value
  ...
  showThisDialog (): void {
    this.whetherToShowThisDialog = true
  }
  ...
  closeThisDialog (): void {
    this.whetherToShowThisDialog = false
  }
  ...
```

I prefer this approach because I don't have to touch any other system files (App.vue, interfaces, store, state, getters, actions). It's all here. And the parent can respond to close events (or you  could implement "exit" or "retry" events instead).

What do you think? There isn't a correct answer but depending on the app and what you're doing in it, there may be a "better" answer :)

Comment (severinbeauvais) [reply src/components/new-request/search.vue:105]: Example: https://github.com/bcgov/business-filings-ui/blob/05b5e3b9950dd5d2f598c66d008394eca99466a5/src/views/AgmLocationChg.vue#L8

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: What did they say about the messages?

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Fixed, thanks.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Changed, thanks.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

File: src/components/new-request/search.vue
Comment (jamespaologarcia) [reply src/components/new-request/search.vue:105]: You're right. Also saves time.

File: src/components/dialogs/societies-info-dialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/societies-info-dialog.vue]: Fixed, thanks.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Fixed, thanks!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

**Comment by severinbeauvais** [comment]
This flow still doesn't work -- should show other input components after I select this society.

image

File: src/components/new-request/search.vue
Comment (jamespaologarcia) [reply src/components/new-request/search.vue]: Checked with Yui Sotozaki. Made the changes accordingly.

Should be "To restore a Society..."

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: I think this will work itself out when the final flow is fixed. Resolving.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Thanks. Resolving.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

**Comment by severinbeauvais** [comment]
> This flow still doesn't work -- should show other input components after I select this society.

Still doesn't work after latest changes.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

File: package-lock.json
Comment (severinbeauvais) [inline package-lock.json:1]: I'm not sure what happened here. The usual procedure is:
1. update app version in package.json
2. enter `npm i`
3. commit package.json and package-lock.json

Comment (jamespaologarcia) [reply package-lock.json:1]: It was a typho on my part. Will fix in the next push

Comment (severinbeauvais) [reply package-lock.json:1]: Resolving.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: I'm resolving this conversation but it might be good reading for the other reviewers.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue]: Thanks for the changes.

PS Instead of specifying a prop as dynamic, which required you to pass the string as an expression, you could do this.

Before:
image

After:
```
  type="request a name for"
```

This works because the string is static.

There's a similar thing when passing boolean props. You can do this (note: no quotes):
```
  myProp=true
```
or this (dynamic prop so what is in quotes is evaluated):
```
  :myProp="true"
```

Comment (severinbeauvais) [reply src/components/new-request/search.vue]: I'm resolving this conversation because your existing code is OK.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

**Comment by JazzarKarim** [comment]
> This flow still doesn't work -- should show other input components after I select this society.
> 
> image

Sev, I've tried this in DEV, it's the same thing:
dev society

I wonder, what do we exactly need to show in that case? For XS, we show the usual jurisdiction and full legal name like you mentioned (if FF is true). In the case of a regular society though, we didn't have previous behavior. The UXPin also didn't have any mention for that.

**Comment by severinbeauvais** [comment]
> Sev, I've tried this in DEV, it's the same thing:
> 
> I wonder, what do we exactly need to show in that case? For XS, we show the usual jurisdiction and full legal name like you mentioned (if FF is true). In the case of a regular society though, we didn't have previous behavior. The UXPin also didn't have any mention for that.

I've been communicating with James on this. It's partly a FF issue and partly a logic issue. See:
https://github.com/bcgov/namerequest/pull/743/commits/4cea5ca5fb9866e744e33cbb3228e154d0cc55a5?diff=split&w=1

**Review by severinbeauvais — APPROVED** [review]
Looks good. Do all your tests pass?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-741-l6xc5zp1.web.app

**Review by JazzarKarim — APPROVED** [review]
LGTM!

**Pull request merged**
